### PR TITLE
[ISSUE #1043] Fix the problem of shared registry singleton.

### PR DESCRIPTION
**Pull request by mytang0**
Fixes ISSUE #1043

### CODE REVIEW

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/registry/Registry.java [hunk 1]
```diff
@@ -31,23 +31,37 @@
...
+            started = true;
+        }
     }
 
-    public void shutdown() throws Exception {
-        registryService.shutdown();
+    public synchronized void shutdown() throws Exception {
+        if (started) {
+            registryService.shutdown();
+            started = false;
```
Comment (li-xiao-shuang) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/registry/Registry.java:63]: inited = false;

**Comment by li-xiao-shuang** [comment]
Register only one protocol service if three protocols are started at the same time?

**Review by li-xiao-shuang — APPROVED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/registry/Registry.java (hunk 1 above)
Comment (mytang0) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/registry/Registry.java:63]: Thanks, fixed.

Comment (mytang0) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/registry/Registry.java:63]: > inited = false;

Thanks, fixed.

**Review by li-xiao-shuang — APPROVED** [review]

**Comment by li-xiao-shuang** [comment]
LGTM

**Comment by qqeasonchen** [comment]
@ruanwenjun please help to check,thanks.

**Pull request merged**